src/stream/src/executor/source/fetch_executor.rs

```diff
@@ -256,16 +266,36 @@ impl<S: StateStore> FsFetchExecutor<S> {
                                 // Receiving file assignments from upstream list executor,
                                 // store into state table and try building a new reader.
                                 Message::Chunk(chunk) => {
-                                    let file_assignment = chunk
-                                        .data_chunk()
-                                        .rows()
-                                        .map(|row| {
-                                            let filename = row.datum_at(0).unwrap().into_utf8();
-                                            let size = row.datum_at(2).unwrap().into_int64();
-                                            FsSplit::new(filename.to_owned(), 0, size as usize)
-                                        })
-                                        .collect();
-                                    state_store_handler.take_snapshot(file_assignment).await?;
+                                    let properties = source_desc.source.config.clone();
+                                    match properties{
+                                        risingwave_connector::source::ConnectorProperties::Gcs(_) => {
+                                            let file_assignment = chunk
+                                            .data_chunk()
+                                            .rows()
+                                            .map(|row| {
+                                                let filename = row.datum_at(0).unwrap().into_utf8();
+                                                let size = row.datum_at(2).unwrap().into_int64();
+                                                    let split: OpendalFsSplit<GcsProperties>= OpendalFsSplit::new(filename.to_owned(), 0, size as usize);
+                                                    split
+                                            })
+                                            .collect();
+                                            state_store_handler.take_snapshot(file_assignment).await?;
+                                        },
+                                        risingwave_connector::source::ConnectorProperties::OpenDalS3(_) => {
+                                            let file_assignment = chunk
+                                            .data_chunk()
+                                            .rows()
+                                            .map(|row| {
+                                                let filename = row.datum_at(0).unwrap().into_utf8();
+                                                let size = row.datum_at(2).unwrap().into_int64();
+                                                    let split: OpendalFsSplit<OpendalS3Properties>= OpendalFsSplit::new(filename.to_owned(), 0, size as usize);
+                                                    split
```

Comment:
dup code

Reply:
We need to specific the type of OpendalFsSplit according to the ConnectorProperties, so it's a bit redundant.
```
gcs => let fs_split: OpendalFsSplit<GCSProperties> = ...
s3 => let fs_split: OpendalFsSplit<OpendalS3Properties> = ...
```